smartgymapi/lib/spotify/spotify.py

```diff
@@ -0,0 +1,120 @@
+import json
+import logging
+import random
+
+import requests
+from pyramid.httpexceptions import HTTPBadRequest
+
+from smartgymapi.models.user import list_users_in_gym
+from smartgymapi.models.music_preference import list_music_preferences_for_users_in_gym
+
+log = logging.getLogger(__name__)
+
+
+class Spotify(object):
+
+    def __init__(self, request, gym):
+        self.request = request
+        self.gym = gym
+        self.settings = request.registry.settings
+        self.access_token = self.get_access_token()
+        self.post_headers = {"Content-Type": "application/json",
+                             "Authorization":
+                             "Bearer {}".format(self.access_token)}
+        self.get_headers = {
+            "Authorization": "Bearer {}".format(self.access_token)}
+
+    def get_access_token(self):
+        """
+        This function gets an acces token which we need to send requests to
+        the spotify API.
+        """
+
+        auth_body = {'grant_type': 'refresh_token',
+                     'refresh_token':
+                     self.settings['spotify.refresh_token'],
+                     'client_id': self.settings['spotify.client.id'],
+                     'client_secret':
+                     self.settings['spotify.client.secret']}
+        r = requests.post(self.settings['spotify.authorize_url'],
+                          data=auth_body)
+        return r.json()['access_token']
+
+    def update_playlist(self):
+        """
+        This function updates the spotify playlist with music based on the
+        users currently in the gym
+        """
+        if not self.gym.spotify_playlist_id:
+            # this means the playlist does not exist yet.
+            # so we have to create it
+
+            r = requests.post(
+                '{}/users/{}/playlists'.format(
+                    self.settings['spotify.base_url'],
+                    self.settings['spotify.user_id']),
+                headers=self.post_headers,
+                data=json.dumps({"name": str(self.gym.id)}))
+
+            if r.status_code is requests.codes.ok:
+                self.gym.spotify_playlist_id = r.json()['id']
+            else:
```

Comment:
Ik denk dat hier nog een scenario bij moet als je access_token verloopt dat je een nieuwe aanvraagt en opnieuw de request maakt. Klopt dat? Checken of die verlopen is is nog mooier dan trial en error toepassen maar misschien wel wat lastiger.